### PR TITLE
Add OCP+ support

### PR DESCRIPTION
**Pull request by jensmueller-com**
This pull request contains the following changes:

- Completely implement the [FYRE OCP+ APIs](https://w3.ibm.com/w3publisher/fyre/ocp/ocp-apis)

### Implementation

- All Click commands invoke a method of an instance of the `dg.lib.fyre.api_manager.OCPPlusAPIManager` class (= API manager).
- For each Click command, the API manager creates an instance of the corresponding request manager (package: `dg.lib.fyre.request_managers`).
- Each request manager executes a REST call and validates the response using the corresponding response manager (package: `dg.lib.fyre.response_managers`).
- Some request managers wrap the JSON response of the REST call in a data object that allows pretty-printing (package: `dg.lib.fyre.data`).

### CODE REVIEW

**Review by choerst-ibm — CHANGES_REQUESTED** [review]
Due to the sheer volume of this PR, I kept most of my comments to usability/documentation topics regarding the CLI functions. I also did one quick pass to look for any functional errors, but this looked good at first glance.

File: dg/commands/fyre/cluster/add_worker_node.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+@click.option("--site", help="OCP+ site", type=click.Choice(["rtp", "svl"]))
+@click.option(
+    "--worker-node-additional-disk-size",
+    callback=validate_worker_node_additional_disk_size,
+    help="Size of additional worker node disk",
+    multiple=True,
+    type=click.IntRange(1, 1000),
+)
+@click.option("--worker-node-count", help="Number of worker nodes", type=click.INT)
+@click.option("--worker-node-num-cpus", help="Number of CPUs per worker node", type=click.IntRange(1, 16))
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/fyre/cluster/add_worker_node.py:55]: There is a variant of this command that already uses the right CPU and RAM values when aiming to use the cluster for Db2 Data Gate: `dg fyre cluster create-for-db2-data-gate`. Therefore, I would like to refrain from overwriting FYRE's default values as any other default values seem arbitrary to me.

File: dg/commands/fyre/cluster/create.py [hunk 2]
```diff
@@ -12,64 +12,180 @@
...
-        "worker_quantity": 3,
-    }
+@optgroup.option("--haproxy-timeout-queue", callback=validate_haproxy_timeout_setting, help="HAProxy queue timeout")
+@optgroup.option("--haproxy-timeout-server", callback=validate_haproxy_timeout_setting, help="HAProxy server timeout")
+@optgroup.option("--ocp_version", callback=validate_ocp_version, help="OpenShift version")
+@optgroup.option("--platform", help="Platform", type=click.Choice(["p", "x", "z"]))
+@optgroup.option("--product-group-id", help="FYRE product group ID", type=click.INT)
+@optgroup.option("--pull-secret", help="Pull secret")
+@optgroup.option("--site", help="OCP+ site", type=click.Choice(["rtp", "svl"]))
+@optgroup.option("--ssh-key", help="SSH key")
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/fyre/cluster/create.py:92]: I slightly changed the help text as the given SSH key is only installed on the infrastructure node.

File: dg/commands/fyre/cluster/create_default.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+from dg.utils.logging import loglevel_command
+
+
+@loglevel_command(
+    context_settings=dg.lib.click.utils.create_default_map_from_json_file(
+        dg.config.data_gate_configuration_manager.get_dg_credentials_file_path()
+    )
+)
+@click.option("--fyre-user-name", help="FYRE API user name", required=True)
+@click.option("--fyre-api-key", help="FYRE API key", required=True)
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/fyre/cluster/create_default.py:33]: I changed the help text to "FYRE API key (https://fyre.svl.ibm.com/account)" in all 32 files specifying this option.

File: dg/commands/fyre/cluster/create_for_db2_data_gate.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+)
+@optgroup.option(
+    "--haproxy-timeout-http-request", callback=validate_haproxy_timeout_setting, help="HAProxy http-request timeout"
+)
+@optgroup.option("--haproxy-timeout-queue", callback=validate_haproxy_timeout_setting, help="HAProxy queue timeout")
+@optgroup.option("--haproxy-timeout-server", callback=validate_haproxy_timeout_setting, help="HAProxy server timeout")
+@optgroup.option("--ocp_version", callback=validate_ocp_version, help="OpenShift version")
+@optgroup.option("--platform", help="Platform", type=click.Choice(["p", "x", "z"]))
+@optgroup.option("--product-group-id", help="FYRE product group ID", type=click.INT)
+@optgroup.option("--pull-secret", help="Pull secret")
```
[the remark this answers is not in the document]
Comment (jensmueller-com) [reply dg/commands/fyre/cluster/create_for_db2_data_gate.py:89]: Unfortunately, I have no idea what the semantics of this option is. The [OCP+ APIs](https://w3.ibm.com/w3publisher/fyre/ocp/ocp-apis) keep schtum about it.

**Comment by jensmueller-com** [comment]
Christian (@choerst-ibm), could you please approve? Thanks!

**Review by choerst-ibm — APPROVED** [review]

**Pull request merged**